pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -98,8 +98,22 @@ type DashboardSpec struct {
 }
 
 type ClusterStatus struct {
-	State   ClusterState `json:"state,omitempty"`
-	Message string       `json:"message,omitempty"`
+	State      ClusterState `json:"state,omitempty"`
+	Message    string       `json:"message,omitempty"`
+	CephStatus *CephStatus  `json:"ceph,omitempty"`
+}
+
+type CephStatus struct {
+	Health         string                       `json:"health,omitempty"`
+	Details        map[string]CephHealthMessage `json:"details,omitempty"`
+	LastChecked    string                       `json:"lastChecked,omitempty"`
+	LastChanged    string                       `json:"lastChanged,omitempty"`
+	PreviousHealth string                       `json:"previousHealth,omitempty"`
+}
+
+type CephHealthMessage struct {
+	Severity string `json:"severity"`
+	Message  string `json:"message"`
```

Comment:
Can we add `detail` too? So that everything can be consumed externally by let's say a UI?

Reply:
The message comes from the detailed message returned from `ceph status --format json` so I believe this is already retrieving all the details. See the PR description for examples of the output. Do you see other details that could be added?

Reply:
My bad, I was using `ceph health details` which returns another field. This is fine.